lldb/include/lldb/API/SBError.h

```diff
@@ -97,7 +97,7 @@ class LLDB_API SBError {
   friend class lldb_private::ScriptInterpreter;
   friend class lldb_private::python::SWIGBridge;
 
-  SBError(const lldb_private::Status &error);
+  SBError(lldb_private::Status &&error);
```

Comment:
This breaks ABI, is this absolutely required?

Reply:
I was operation under the assumption that this is an implementation detail. This is a method that takes an `lldb_private`(!) type as parameter.
@bulbazord What use-case would this break?

Reply:
I have no concerns about Unix-ish platforms, but Windows may be a problem. I know that MSVC's mangling scheme includes access level (private/protected/public), which makes me think that private/protected symbols may be difficult to change while keeping ABI stable on MSVC.
Certainly SBAPI users and libraries shouldn't even be able to use this symbol, but I don't know if Windows will be unhappy that a private symbol changed.

@compnerd Do you have any insight here?

Reply:
I agree with @JDevlieghere that we shouldn't consider private symbols as ABI. As long as we do not decrease the access level, I think that the breakage should be fine.

Reply:
+1

Reply:
Changing the argument type will also change the mangled name, so I don't think that the access level matters here (what this is really doing is adding a new function and (simultaneously) removing an existing one. However, I don't think that should be a problem, since people shouldn't be using the lldb_private functions.